### ISSUE TITLE
Suggests param typehint

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#834]
Reading the `erusev/parsedown` repository [**I see that code**](https://github.com/erusev/parsedown/blob/6678d59be48c4be64eaca6ce70bea48a09488cc2/Parsedown.php#L80):

```php
    function setSafeMode($safeMode)
    {
        $this->safeMode = (bool) $safeMode;
        return $this;
    }
```

This just uses the param `$safeMode` to cast it to `(bool)` which visibly was done to keep compatibility with PHP versions that does not support param typehint. But if my project does support it (PHP 7+), then it could be avoided by expect the right parameter type.

```php
    function setSafeMode(bool $safeMode)
    {
        $this->safeMode = $safeMode;
        return $this;
    }
```

**Comment by kalessil** [comment]
Note: ArrayTypeOfParameterByDefaultValueInspection should go away when this is implemented

**Comment by funivan** [comment]
My case:
```php
function show($names) : void {
  shuffle($names);
  echo reset($names);
}

```
We can suggest to use array in parameter types

```php
function show(array $names) : void {
  shuffle($names);
  echo reset($names);
}

```

**Comment by kalessil** [comment]
Update: ArrayTypeOfParameterByDefaultValueInspection was recently dropped

**Comment by rentalhost** [comment]
I don't know if it is a good suggestion, now. Because it will force `setSafeMode()` accepts only boolean as argument. So other castable values will fail if the suggestion is applied.

Before:

```php
$this->setSafeMode($notBooleanVariable); // maybe fails
```

**Comment by ea-inspections-team** [comment]
Thanks for reporting @rentalhost ! Looks like PhpStorm already implemented this feature. Closing the issue.
